Assets/VIRTUOSO/VRTK/Source/SDK/Unity/SDK_UnityController.cs

```diff
@@ -138,15 +210,101 @@ Axis Codes
         /// <param name="options">A dictionary of generic options that can be used to within the update.</param>
         public override void ProcessUpdate(VRTK_ControllerReference controllerReference, Dictionary<string, object> options)
         {
+#if UNITY_2019_1_OR_NEWER
+            if (controllerReference.hand == ControllerHand.Left)
+            {
+                ProcessUnityXRInput(leftHandCurrentFrameButtonStates, leftControllerXRDevice);
+            }
+            else
+            {
+                ProcessUnityXRInput(rightHandCurrentFrameButtonStates, rightControllerXRDevice);
+            }
+#endif
         }
 
+#if UNITY_2019_1_OR_NEWER
+        /// <summary>
+        /// Uses Unity XR input to determine the status of each button on the controller.
+        /// </summary>
+        protected void ProcessUnityXRInput(Dictionary<ButtonTypes,UnityXRInputState> handState, InputDevice currentController)
+        {
+            if(currentController.isValid != true)
+            {
+                return;
+            }
+
+            // get current button values
+            foreach (ButtonTypes buttonType in handState.Keys)
+            {
+                bool lastFrameTouchValue = handState[buttonType].buttonState[ButtonPressTypes.Touch];
+                bool lastFramePressValue = handState[buttonType].buttonState[ButtonPressTypes.Press];
+
+                bool currentFrameTouchValue = false;
+                bool currentFramePressValue = false;
+                bool buttonValueOut;
+                float floatValueOut = 0.0f;
```

Comment:
no need for "float" in the variable name. Also, specific on what the value is out?